extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts

```diff
@@ -282,4 +263,34 @@ export class RemoteQueriesManager extends DisposableObject {
     const filePath = path.join(this.storagePath, queryItem.queryId);
     return await fs.pathExists(filePath);
   }
+
+  /**
+   * Checks whether there's a result index artifact available for the given query.
+   * If so, set the query status to `Completed` and auto-download the results.
+   */
+  private async downloadAvailableResults(queryItem: RemoteQueryHistoryItem, credentials: Credentials, executionEndTime: number): Promise<void> {
+    const resultIndex = await getRemoteQueryIndex(credentials, queryItem.remoteQuery);
+    if (resultIndex) {
+      queryItem.completed = true;
+      queryItem.status = QueryStatus.Completed;
+      queryItem.failureReason = undefined;
+      const queryResult = this.mapQueryResult(executionEndTime, resultIndex, queryItem.queryId);
+
+      await this.storeJsonFile(queryItem, 'query-result.json', queryResult);
+
+      // Kick off auto-download of results in the background.
+      void commands.executeCommand('codeQL.autoDownloadRemoteQueryResults', queryResult);
+
+      // Ask if the user wants to open the results in the background.
+      void this.askToOpenResults(queryItem.remoteQuery, queryResult).then(
+        noop,
+        err => {
+          void showAndLogErrorMessage(err);
+        }
+      );
+    } else {
+      void showAndLogErrorMessage(`There was an issue retrieving the result for the query ${queryItem.remoteQuery.queryName}`);
```

Comment:
Since we know what the URL of the workflow is, maybe we should add it here. (Not exactly part of this PR, but still a nice to have.)